src/polyfem/solver/forms/Form.hpp

```diff
@@ -111,7 +111,7 @@ namespace polyfem::solver
 		bool enabled() const { return enabled_; }
 
 		/// @brief Get the form's multiplicative constant weight
-		double weight() const { return weight_; }
+		virtual double weight() const { return weight_; }
```

Comment:
i don't like this change, it kinds of break the whole automatic weighting system

Reply:
I was tempted to scrap the whole automatic weighting thing and make it explicit in each form but I decided not to... It just becomes confusing when the same variable (`weight_`) encodes many different things (barrier stiffness, acceleration scaling, AL weight, regularization weight) and some of them don't even have a weight (`InertiaForm` and `BCLagrangianForm`). I looked through the code and couldn't find an instance where it was necessary to compute the value/grad without the weighting.